### PR TITLE
chore: Player class removal

### PR DESCRIPTION
**Pull request by EmosewaMC**
Commits tell a better story.  **Please reference those for testing and changes**.  Changes were done incrementally in chunks so as to ensure isolated testing cases and functional changes.

Tested that I cannot cheat pick up dropped loot or coins still and that GetSystemAddress calls are correct and good.

### CODE REVIEW

File: dGame/dUtilities/SlashCommandHandler.cpp [hunk 1]
```diff
@@ -1109,10 +1108,10 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& command, Entit
 					return;
 				}
 			} else {
-				accountId = player->GetParentUser()->GetAccountID();
+				if (player->GetCharacter() && player->GetCharacter()->GetParentUser()) accountId = player->GetCharacter()->GetParentUser()->GetAccountID();
```
Comment (Xiphoseer) [inline dGame/dUtilities/SlashCommandHandler.cpp:1111]: In general you should use the pointer you checked against null, not the next pointer from the function that is side-effect free in theory

File: dWorldServer/WorldServer.cpp [hunk 2]
```diff
@@ -1127,9 +1126,8 @@ void HandlePacket(Packet* packet) {
 				//Mail::HandleNotificationRequest(packet->systemAddress, player->GetObjectID());
 
 				//Notify chat that a player has loaded:
-				{
+				if (player->GetCharacter() && player->GetCharacter()->GetParentUser()) {
```
Comment (Xiphoseer) [inline dWorldServer/WorldServer.cpp:1129]: Same here

File: dWorldServer/WorldServer.cpp [hunk 3]
```diff
@@ -608,8 +607,8 @@ void HandlePacketChat(Packet* packet) {
 
 				auto* entity = Game::entityManager->GetEntity(playerId);
 
-				if (entity != nullptr) {
-					entity->GetParentUser()->SetMuteExpire(expire);
+				if (entity && entity->GetCharacter() && entity->GetCharacter()->GetParentUser()) {
```
Comment (Xiphoseer) [inline dWorldServer/WorldServer.cpp:610]: Same here

File: dGame/dUtilities/SlashCommandHandler.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dGame/dUtilities/SlashCommandHandler.cpp:1111]: these getters are inlined to just access the field but alright

**Review by Xiphoseer — APPROVED** [review]

File: dGame/dUtilities/SlashCommandHandler.cpp (hunk 1 above)
Comment (EmosewaMC) [reply dGame/dUtilities/SlashCommandHandler.cpp]: wasnt any snide remark meant here.  May have come off as that, sorry.

**Review by aronwk-aaron — APPROVED** [review]

**Pull request merged**